consumer/consumererror/networkerrors.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+import (
+	"fmt"
+	"net/http"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"go.opentelemetry.io/collector/consumer/consumererror/internal/statusconversion"
+)
+
+// NetworkError contains either an HTTP or gRPC status code
+// resulting from a failed synchronous network request.
+// It allows retrieving a type of error code, converting
+// between the status codes supported by each transport if necessary.
+//
+// It should be created with NewHTTPStatus or NewGRPCStatus.
+type NetworkError struct {
+	error
```

Comment:
Similar to Partial, may be better `err error`

Reply:
Sounds good, done.